# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [3]:
import numpy as np
import pandas as pd
from scipy import stats
from statsmodels.stats import weightstats as stests

In [17]:
data = 128, 135, 121, 142, 126, 151, 123
mean_in_commerce = 123.7
alpha = 0.10
alpha_2 = 0.05

#Завдання № 1

Гіпотези6
- Но : 𝜇0=123.7
- Н1 : 𝜇1>123.7
- Гіпотеза правостороня

#Завдання № 2

In [9]:
x_mean = np.mean(data)
s = np.std(data, ddof=1)
n = len(data)
print(x_mean, s, n)

132.28571428571428 10.980502200328935 7


#Завдання № 3

In [20]:
d_of_free = n - 1
t_crit = stats.t.ppf(1-alpha, d_of_free)

t_stat = (x_mean - mean_in_commerce) / (s / np.sqrt(n))

p_value = 1 - stats.t.cdf(t_stat, d_of_free)

print(f't-критичне: {t_crit:.3f}')
print(f't-статистика: {t_stat:.3f}')
print(f'p-value: {p_value:.4f}')

if p_value < alpha:
    print(f"Відхиляємо H0: середній час до покупки збільшився на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

t-критичне: 1.440
t-статистика: 2.069
p-value: 0.0420
Відхиляємо H0: середній час до покупки збільшився на рівні значущості 0.1


In [14]:
res = stats.ttest_1samp(data, popmean=mean_in_commerce, alternative='greater')
print(f'scipy: t-статистика = {res.statistic:.3f}, p-value = {res.pvalue:.5f}')

scipy: t-статистика = 2.069, p-value = 0.04202


#Висновок
- t-статистика > t-критичне (2.069 > 1.440)
- p-value менше за альфа (0.0420 < 0.10)
- відхиляємо гіпотезу Но
- є статистично значуще збільшення часу до покупки після редизайну на рівні значущості 0.10

# Завдання № 6

In [18]:
d_of_free = n - 1
t_crit = stats.t.ppf(1-alpha_2, d_of_free)

t_stat = (x_mean - mean_in_commerce) / (s / np.sqrt(n))

p_value = 1 - stats.t.cdf(t_stat, d_of_free)

print(f't-критичне: {t_crit:.3f}')
print(f't-статистика: {t_stat:.3f}')
print(f'p-value: {p_value:.4f}')

t-критичне: 1.943
t-статистика: 2.069
p-value: 0.0420


Оскільки p-value (0.0420) значно менше за альфа, висновок не зміниться навіть при більш жорсткому рівні значущості 0.05. Це підтверджує наявність статистично значущої різниці у часі до покупки після редизайну.

Також, це підтверджується співідношенням t-статистика > t-критичне (2.069 > 1.943)

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

гіпотези
-Но: р=р0
- Н1: р>р0

In [23]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count = 311   # кількість людей, що впізнали логотип
nobs = 1200    # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print(f'Z-статистика = {stat:.3f}')
print(f'p-value = {pval:.4f}')
print(f'97% довірчий інтервал: {confint}')

Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


###Висновок:
-рекламна кампанія була ефективною — частка дорослих, які впізнають логотип компанії, статистично значуще зросла (p < 0.03) і становить більше 23%.

- з довірою 97% (альфа = 0.03), стверджувати, що справжня частка всіх дорослих, які впізнають логотип компанії після кампанії відділу маркетингу, знаходиться між 23.17% та 28.66%. Довірчий інтервал не містить значення 0.23, що додатково підтримує висновок про зростання впізнаваності логотипу.